segmentstore/server/src/main/java/io/pravega/segmentstore/server/logs/OperationProcessor.java

```diff
@@ -264,24 +253,24 @@ private void processOperations(Queue<CompletableOperation> operations) {
                     if (operations.isEmpty()) {
                         log.debug("{}: processOperations (Flush).", this.traceObjectId);
                         synchronized (this.stateLock) {
-                            builder.flush();
+                            this.dataFrameBuilder.flush();
                         }
                     } else {
                         log.debug("{}: processOperations (Add OperationCount = {}).", this.traceObjectId, operations.size());
                     }
                 }
             } catch (Throwable ex) {
                 // Fail ALL the operations that haven't been acknowledged yet.
-                Throwable realCause = ExceptionHelpers.getRealException(ex);
-                this.state.fail(realCause, null);
+                ex = ExceptionHelpers.getRealException(ex);
+                this.state.fail(ex, null);
 
-                if (isFatalException(realCause)) {
+                if (isFatalException(ex)) {
                     // If we encountered a fatal exception, it means we detected something that we cannot possibly recover from.
                     // We need to shutdown right away (this will be done by the main loop).
 
                     // But first, fail any Operations that we did not have a chance to process yet.
-                    cancelIncompleteOperations(operations, realCause);
-                    throw Lombok.sneakyThrow(realCause);
+                    cancelIncompleteOperations(operations, ex);
+                    throw Lombok.sneakyThrow(ex);
```

Comment:
I know this is not changing from the previous code, but I want to understand why you are sneaky throwing it here rather than simply throwing.